libarchive/archive_read_support_format_rar5.c

```diff
@@ -4059,6 +4082,10 @@ static int rar5_read_data(struct archive_read *a, const void **buff,
 	if (size)
 		*size = 0;
 
+	if (rar->has_encrypted_entries == ARCHIVE_READ_FORMAT_ENCRYPTION_DONT_KNOW) {
+		rar->has_encrypted_entries = 0;
+	}
```

Comment:
I wanted to add an `else` clause here to early return failure w/ an accurate and useful error message if the entry is encrypted or if the archive is solid and encryption has been identified. When trying this out, I ran into issues elsewhere that assumed at least some of this code path has run and done some initialization. Looking elsewhere, I also didn't really see other formats doing such a check - I know that for RAR4 we attempt to read normally and typically fail when things don't work out properly and that for 7zip files there's no specific code-path for returning errors. If folks think that early-returning here is beneficial, I can try and pursue this again.

Reply:
Actually, on second thought, I probably want to address the other errors I was getting anyway. If I remove the attempt to `archive_read_data` in the tests for the first entry, the solid archive test starts failing with the same error message as I was getting when trying to early return: `Declared solid file, but no window buffer initialized yet.`

I'll either need to ensure that the window buffer is initialized in the skip code or make the check raising this error less strict and understand the notion of encrypted solid archives.